sdk/src/wallet/account/operations/balance.rs

```diff
@@ -30,108 +34,98 @@ impl AccountHandle {
 
         let local_time = self.client.get_time_checked().await?;
 
-        let mut total_amount = 0;
         let mut total_rent_amount = 0;
-        let mut required_storage_deposit = RequiredStorageDeposit::new();
         let mut total_native_tokens = NativeTokensBuilder::new();
-        let mut potentially_locked_outputs = HashMap::new();
-        let mut aliases = Vec::new();
-        let mut foundries = Vec::new();
-        let mut nfts = Vec::new();
 
         let account = self.read().await;
 
-        for output_data in account.unspent_outputs.values() {
+        let relevant_unspent_outputs = account
+            .unspent_outputs
+            .values()
             // Check if output is from the network we're currently connected to
-            if output_data.network_id != network_id {
-                continue;
-            }
+            .filter(|data| data.network_id == network_id)
+            .map(|data| (&data.output_id, &data.output));
```

Comment:
filter_map()?

Reply:
Sure, weird that clippy didn't find this

Reply:
Thoralf > Clippy

Reply:
Hmm actually not sure, you need something that returns an option for filter_map